controllers/nonadmin_controller.go

```diff
@@ -0,0 +1,205 @@
+		"app.kubernetes.io/component":  "manager",
+		"app.kubernetes.io/created-by": common.OADPOperator,
+		"app.kubernetes.io/instance":   nonAdminObjectName,
+		"app.kubernetes.io/managed-by": "kustomize",
+		"app.kubernetes.io/name":       "deployment",
+		"app.kubernetes.io/part-of":    common.OADPOperator,
+	}
+)
+
+func (r *DPAReconciler) ReconcileNonAdminController(log logr.Logger) (bool, error) {
+	// TODO https://github.com/openshift/oadp-operator/pull/1316
+	dpa := oadpv1alpha1.DataProtectionApplication{}
+	if err := r.Get(r.Context, r.NamespacedName, &dpa); err != nil {
+		return false, err
+	}
+
+	nonAdminDeployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      nonAdminObjectName,
+			Namespace: r.NamespacedName.Namespace,
+		},
+	}
+
+	// Delete (possible) previously deployment
+	if !r.checkNonAdminEnabled(&dpa) {
+		if err := r.Get(
+			r.Context,
+			types.NamespacedName{
+				Name:      nonAdminDeployment.Name,
+				Namespace: nonAdminDeployment.Namespace,
+			},
+			nonAdminDeployment,
+		); err != nil {
+			if k8serror.IsNotFound(err) {
+				return true, nil
+			}
+			return false, err
+		}
+
+		deleteOptionPropagationForeground := metav1.DeletePropagationForeground
+		if err := r.Delete(
+			r.Context,
+			nonAdminDeployment,
+			&client.DeleteOptions{PropagationPolicy: &deleteOptionPropagationForeground},
+		); err != nil {
+			r.EventRecorder.Event(
+				nonAdminDeployment,
+				corev1.EventTypeWarning,
+				"NonAdminDeploymentDeleteFailed",
+				fmt.Sprintf("Could not delete non admin controller deployment %s/%s: %s", nonAdminDeployment.Namespace, nonAdminDeployment.Name, err),
+			)
+			return false, err
+		}
+		r.EventRecorder.Event(
+			nonAdminDeployment,
+			corev1.EventTypeNormal,
+			"NonAdminDeploymentDeleteSucceed",
+			fmt.Sprintf("Non admin controller deployment %s/%s deleted", nonAdminDeployment.Namespace, nonAdminDeployment.Name),
+		)
+		return true, nil
+	}
+
+	operation, err := controllerutil.CreateOrUpdate(
+		r.Context,
+		r.Client,
+		nonAdminDeployment,
+		func() error {
+			err := r.buildNonAdminDeployment(nonAdminDeployment, &dpa)
+			if err != nil {
+				return err
+			}
+
+			// Setting controller owner reference on the non admin controller deployment
+			return controllerutil.SetControllerReference(&dpa, nonAdminDeployment, r.Scheme)
+		},
+	)
+	if err != nil {
+		return false, err
+	}
+
+	if operation != controllerutil.OperationResultNone {
+		r.EventRecorder.Event(
+			nonAdminDeployment,
+			corev1.EventTypeNormal,
+			"NonAdminDeploymentReconciled",
+			fmt.Sprintf("Non admin controller deployment %s/%s %s", nonAdminDeployment.Namespace, nonAdminDeployment.Name, operation),
+		)
+	}
+	return true, nil
+}
+
+func (r *DPAReconciler) buildNonAdminDeployment(deploymentObject *appsv1.Deployment, dpa *oadpv1alpha1.DataProtectionApplication) error {
+	// TODO https://github.com/openshift/oadp-operator/pull/1316
+	nonAdminImage := r.getNonAdminImage(dpa)
```

Comment:
Lets move this function call to `ensureRequiredSpecs` func and modify the `ensureRequiredSpecs` function to just accept the `deploymentObject` as the only parmeter.

Reply:
That would need to make `ensureRequiredSpecs` a method of `DPAReconciler`, because today, get image function needs DPA

I believe is better to be like this 

In the end, we only call `buildNonAdminDeployment` function, and after [Sachin's PR](https://github.com/openshift/oadp-operator/pull/1316) that function will accept the `deploymentObject` as the only parmeter